acceptance/tests/connect/connect_inject_test.go

```diff
@@ -18,25 +18,130 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-// Test that Connect works in a default and a secure installation.
+// TestConnectInject tests that Connect works in a default and a secure installation.
 func TestConnectInject(t *testing.T) {
-	cases := []struct {
+	cases := map[string]struct {
+		clusterKind consul.ClusterKind
+		releaseName string
 		secure      bool
 		autoEncrypt bool
 	}{
-		{false, false},
-		{true, false},
-		{true, true},
+		"Helm install without secure or auto-encrypt": {
+			clusterKind: consul.Helm,
+			releaseName: helpers.RandomName(),
+		},
+		"Helm install with secure": {
+			clusterKind: consul.Helm,
+			releaseName: helpers.RandomName(),
+			secure:      true,
+		},
+		"Helm install with secure and auto-encrypt": {
+			clusterKind: consul.Helm,
+			releaseName: helpers.RandomName(),
+			secure:      true,
+			autoEncrypt: true,
+		},
+		"CLI install without secure or auto-encrypt": {
+			clusterKind: consul.CLI,
+			releaseName: consul.CLIReleaseName,
+		},
+		"CLI install with secure": {
+			clusterKind: consul.CLI,
+			releaseName: consul.CLIReleaseName,
+			secure:      true,
+		},
+		"CLI install with secure and auto-encrypt": {
+			clusterKind: consul.CLI,
+			releaseName: consul.CLIReleaseName,
+			secure:      true,
+			autoEncrypt: true,
+		},
 	}
 
-	for _, c := range cases {
-		name := fmt.Sprintf("secure: %t; auto-encrypt: %t", c.secure, c.autoEncrypt)
+	for name, c := range cases {
+		t.Run(name, func(t *testing.T) {
+			cfg := suite.Config()
+			ctx := suite.Environment().DefaultContext(t)
+
+			connHelper := ConnectHelper{
+				ClusterKind: c.clusterKind,
+				Secure:      c.secure,
+				AutoEncrypt: c.autoEncrypt,
+				ReleaseName: c.releaseName,
+				Ctx:         ctx,
+				Cfg:         cfg,
+			}
+
+			connHelper.Setup(t)
+
+			connHelper.Install(t)
+			connHelper.DeployClientAndServer(t)
+			connHelper.TestConnectionFailureWithoutIntention(t)
+			connHelper.CreateIntention(t)
```

Comment:
Can we extract the `if c.secure()` logic from inside create intention to outside it?
ie 
if c.secure {
  connHepler.CreateIntention(t)
}

I assumed that we were always creating intentions when I initially read it.

Reply:
Similarly, move the above assertion about connection failure without intention as well. It is an assertion that only runs when secure is true, but the way it is structured, it feels like it always runs that assertion.

Reply:
I thought about the same thing too and was on the fence. Looking at it again, I think you are right on this.

Reply:
yeah! my rule of thumb is usually to frontload any information that someone working with this code might need to know without having to dive into the method definition or differently put, avoid surprises.